cmd/rollapp/da/switch.go

```diff
@@ -0,0 +1,153 @@
+				WithDefaultText("select the DA you want to switch").
+				WithOptions(envs).
+				Show()
+
+			err := initconfig.AddFlags(cmd)
+			if err != nil {
+				pterm.Error.Println("failed to switch DA: ", err)
+				return
+			}
+
+			home, err := filesystem.ExpandHomePath(
+				cmd.Flag(initconfig.GlobalFlagNames.Home).Value.String(),
+			)
+			if err != nil {
+				pterm.Error.Println("failed to switch DA: ", err)
+				return
+			}
+
+			rollappConfig, err := roller.LoadConfig(home)
+			if err != nil {
+				pterm.Error.Println("failed to load roller config: ", err)
+				return
+			}
+
+			if strings.ToLower(string(rollappConfig.DA.Backend)) == env {
+				pterm.Info.Println("You are switching to the same DA as previous!")
+				return
+			}
+
+			var dalayer datalayer.DataLayer
+			switch env {
+			case "avail":
+				raResp, err := rollapp.GetMetadataFromChain(rollappConfig.RollappID, rollappConfig.HubData)
+				if err != nil {
+					pterm.Error.Println("failed to get metadata from chain: ", err)
+					return
+				}
+
+				drsVersion, err := genesis.GetDrsVersionFromGenesis(home, raResp)
+				if err != nil {
+					pterm.Error.Println("failed to get drs version from genesis: ", err)
+					return
+				}
+
+				drsVersionInt, err := strconv.ParseInt(drsVersion, 10, 64)
+				if err != nil {
+					pterm.Error.Println("failed to get drs version from genesis: ", err)
+					return
+				}
+
+				if !rollapp.IsDaConfigNewFormat(drsVersionInt, strings.ToLower(raResp.Rollapp.VmType)) {
+					pterm.Error.Println("required Rollapp DRS version of at least", drsVersionInt)
+					return
+				}
+
+				dalayer = avail.NewAvail(home)
+
+			default:
+				pterm.Error.Println("switch does not support da: ", env)
+				return
+			}
+
+			submited, _ := pterm.DefaultInteractiveConfirm.WithDefaultText("Have you submitted to gov yet?").Show()
+			if !submited {
+				// Create Gov
+				keyName, _ := pterm.DefaultInteractiveTextInput.WithDefaultText("Enter your key name").Show()
+				keyring, _ := pterm.DefaultInteractiveTextInput.WithDefaultText("Enter your keyring-backend").Show()
+				title, _ := pterm.DefaultInteractiveTextInput.WithDefaultText("Enter your Proposal Title").Show()
+				description, _ := pterm.DefaultInteractiveTextInput.WithDefaultText("Enter your Proposal Description").Show()
+				deposit, _ := pterm.DefaultInteractiveTextInput.WithDefaultText("Enter your Proposal Deposit").Show()
+				newDAParam := json.RawMessage(fmt.Sprintf(`"%s"`, env))
+				txHash, err := gov.ParamChangeProposal(home, keyName, keyring,
+					&utils.ParamChangeProposalJSON{
+						Title:       title,
+						Description: description,
+						Changes: utils.ParamChangesJSON{
+							utils.NewParamChangeJSON("rollappparams", "da", newDAParam),
+						},
+						Deposit: deposit + rollappConfig.Denom,
+					})
+
+				if err != nil {
+					pterm.Error.Println("failed to submit proposal", err)
+					return
+				}
+				pterm.Info.Println("Proposal Tx hash: ", txHash)
```

Comment:
over all good, but there are too many inputs for the user to provide, might need ux improvements in the future